File: class/class4.1/src/Pizza.cpp
#include "Pizza.hpp"

void Pizza::addIngredient(Ingredient&& ingredient) {
    this->ingredients.push_back(std::move(ingredient));
}

std::vector<Ingredient> Pizza::getIngredients() const {
    return this->ingredients;
}

double Pizza::getTotalCost() const {
    double total = 0.0;

    for (const Ingredient& ingredient : this->ingredients) {
        total += ingredient.getCost();
    }
    
    return total;
}

File: home/home4.1/pizza-builder/src/pizza.rs
use crate::ingredient::Ingredient;

pub struct Pizza {
    ingredients: Vec<Ingredient>,
}

impl Pizza {
    pub fn new() -> Self {
        Self {
            ingredients: Vec::new(),
        }
    }

    pub fn add_ingredient(&mut self, ingredient: Ingredient) {
        self.ingredients.push(ingredient);
    }

    pub fn get_ingredients(&self) -> &Vec<Ingredient> {
        &self.ingredients
    }

    pub fn get_total_cost(&self) -> f64 {
        self.ingredients.iter().map(|i| i.get_cost()).sum()
    }
}
